linodecli/configuration.py

```diff
@@ -366,19 +355,32 @@ def do_GET(self):
 <h2>Success</h2><br/><p>You may return to your terminal to continue..</p>
 <script>
 // this is gross, sorry
-let r = new XMLHttpRequest('http://localhost:8123');
+let r = new XMLHttpRequest('http://localhost:{port}');
 r.open('GET', '/token/'+window.location.hash.substr(1));
 r.send();
 </script>
-""", "utf-8"))
+""".format(port=self.server.server_address[1]), "utf-8"))
 
             def log_message(self, form, *args):
                 """Don't actually log the request"""
 
         # start a server to catch the response
-        serv = server.HTTPServer(("localhost",8123), Handler)
+        serv = server.HTTPServer(("localhost",0), Handler)
         serv.token = None
 
+        # figure out the URL to direct the user to and print out the prompt
+        url = "https://login.linode.com/oauth/authorize?client_id={}&response_type=token&scopes=*&redirect_uri=http://localhost:{}".format(
+            OAUTH_CLIENT_ID, serv.server_address[1]
+        )
+        print("""Your browser will be directed to this URL to authenticate:
+
+{}
+
+If you are not automatically directed there, please copy/paste the link into your browser
```

Comment:
we should add a `[Q]` command to abort this and get back to the original prompt (where a user can then choose `t` for token.

Reply:
I don't know if we can - the thread is blocked waiting for a request to come into the server.  Ctr+C should still raise a KeboardInterrupt and exit nicely (I handle it below).